minifi-bootstrap/src/main/java/org/apache/nifi/minifi/bootstrap/util/ConfigTransformer.java

```diff
@@ -0,0 +1,566 @@
+    public static final String SNAPSHOT_FREQUENCY_KEY = "snapshot frequency";
+
+    public static final String SECURITY_PROPS_KEY = "Security Properties";
+    public static final String KEYSTORE_KEY = "keystore";
+    public static final String KEYSTORE_TYPE_KEY = "keystore type";
+    public static final String KEYSTORE_PASSWORD_KEY = "keystore password";
+    public static final String KEY_PASSWORD_KEY = "key password";
+    public static final String TRUSTSTORE_KEY = "truststore";
+    public static final String TRUSTSTORE_TYPE_KEY = "truststore type";
+    public static final String TRUSTSTORE_PASSWORD_KEY = "truststore password";
+    public static final String SENSITIVE_PROPS_KEY = "Sensitive Props";
+    public static final String SENSITIVE_PROPS_KEY__KEY = "key";
+    public static final String SENSITIVE_PROPS_ALGORITHM_KEY = "algorithm";
+    public static final String SENSITIVE_PROPS_PROVIDER_KEY = "provider";
+
+    public static final String PROCESSOR_CONFIG_KEY = "Processor Configuration";
+    public static final String CLASS_KEY = "class";
+    public static final String SCHEDULING_PERIOD_KEY = "scheduling period";
+    public static final String PENALIZATION_PERIOD_KEY = "penalization period";
+    public static final String SCHEDULING_STRATEGY_KEY = "scheduling strategy";
+    public static final String RUN_DURATION_NANOS_KEY = "run duration nanos";
+    public static final String AUTO_TERMINATED_RELATIONSHIPS_LIST_KEY = "auto-terminated relationships list";
+
+    public static final String PROCESSOR_PROPS_KEY = "Properties";
+
+    public static final String CONNECTION_PROPS_KEY = "Connection Properties";
+    public static final String MAX_WORK_QUEUE_SIZE_KEY = "max work queue size";
+    public static final String MAX_WORK_QUEUE_DATA_SIZE_KEY = "max work queue data size";
+    public static final String FLOWFILE_EXPIRATION__KEY = "flowfile expiration";
+    public static final String QUEUE_PRIORITIZER_CLASS_KEY = "queue prioritizer class";
+
+    public static final String REMOTE_PROCESSING_GROUP_KEY = "Remote Processing Group";
+    public static final String URL_KEY = "url";
+    public static final String TIMEOUT_KEY = "timeout";
+
+    public static final String INPUT_PORT_KEY = "Input Port";
+    public static final String USE_COMPRESSION_KEY = "use compression";
+
+    public static final String PROVENANCE_REPORTING_KEY = "Provenance Reporting";
+    public static final String DESTINATION_URL_KEY = "destination url";
+    public static final String PORT_NAME_KEY = "port name";
+    public static final String ORIGINATING_URL_KEY = "originating url";
+    public static final String BATCH_SIZE_KEY = "batch size";
+
+    public static final String SSL_PROTOCOL_KEY = "ssl protocol";
+
+    // Final util classes should have private constructor
+    private ConfigTransformer() {}
+
+    public static void transformConfigFile(String sourceFile, String destPath) throws Exception {
```

Comment:
this method could likely feed into the other transformConfigFile(InputStream, String).